jq_array_parser.go

```diff
@@ -2,43 +2,46 @@ package opslevel_jq_parser
 
 import (
 	"encoding/json"
-	"strings"
+	"github.com/opslevel/opslevel-jq-parser/v2024/common"
 )
 
 type JQArrayParser struct {
-	programs []*JQFieldParser
+	programs []JQFieldParser
 }
 
-func NewJQArrayParser(expressions []string) *JQArrayParser {
-	programs := make([]*JQFieldParser, len(expressions))
+func NewJQArrayParser(expressions []string) JQArrayParser {
+	programs := make([]JQFieldParser, len(expressions))
 	for i, expression := range expressions {
 		programs[i] = NewJQFieldParser(expression)
 	}
-	return &JQArrayParser{
+	return JQArrayParser{
 		programs: programs,
 	}
 }
 
-func (p *JQArrayParser) Run(data string) []string {
-	output := make([]string, 0, len(p.programs))
+func (p JQArrayParser) Run(data string) []string {
+	output := make(common.UniqueMap[bool])
 	for _, program := range p.programs {
 		response := program.Run(data)
 		if response == "" {
 			continue
 		}
-		if strings.HasPrefix(response, "[") && strings.HasSuffix(response, "]") {
-			var aliases []string
```

Comment:
renamed aliases to elements, since this could technically handle things other than aliases (any non-object array type would work)